superset-frontend/src/explore/components/DatasourcePanel.tsx

```diff
@@ -0,0 +1,197 @@
+    margin-bottom: ${({ theme }) => theme.gridUnit * 3}px;
+  }
+  .ant-collapse-item {
+    background-color: ${({ theme }) => theme.colors.grayscale.light4};
+    .anticon.anticon-right.ant-collapse-arrow > svg {
+      transform: rotate(90deg) !important;
+    }
+  }
+  .ant-collapse-item.ant-collapse-item-active {
+    .anticon.anticon-right.ant-collapse-arrow > svg {
+      transform: rotate(-90deg) !important;
+    }
+  }
+  .header {
+    font-size: ${({ theme }) => theme.typography.sizes.l}px;
+    margin-left: ${({ theme }) => theme.gridUnit * -2}px;
+  }
+  .ant-collapse-content-box > div {
+    margin-left: -14px;
+  }
+  .type-label {
+    text-align: left;
+  }
+  .metric-option .option-label {
+    margin-left: ${({ theme }) => theme.gridUnit * -5}px;
+  }
+  .field-selections {
+    position: absolute;
+    top: ${({ theme }) => theme.gridUnit * 15}px;
+    bottom: 0;
+    left: 0;
+    right: 0;
+    overflow: auto;
+  }
+  .field-length {
+    margin-top: -3px;
+    margin-bottom: ${({ theme }) => theme.gridUnit * 2}px;
+    font-size: ${({ theme }) => theme.typography.sizes.s}px;
+    color: ${({ theme }) => theme.colors.grayscale.light1};
+  }
+`;
+
+const DataSourcePanel = ({
+  datasource,
+  controls: { datasource: datasourceControl },
+  actions,
+}: Props) => {
+  const { columns, metrics } = datasource;
+  const [lists, setList] = useState({
+    columns,
+    metrics,
+  });
+  const search = ({ target: { value } }: { target: { value: string } }) => {
+    if (value === '') {
+      setList({ columns, metrics });
+      return;
+    }
+    const filteredColumns = lists.columns.filter(
+      column => column.column_name.indexOf(value) !== -1,
+    );
+    const filteredMetrics = lists.metrics.filter(
+      metric => metric.metric_name.indexOf(value) !== -1,
+    );
+    setList({ columns: filteredColumns, metrics: filteredMetrics });
+  };
+  useEffect(() => {
+    setList({
+      columns,
+      metrics,
+    });
+  }, [datasource]);
+
+  const metricSlice = lists.metrics.slice(0, 50);
+  const columnSlice = lists.columns.slice(0, 50);
+
+  return (
+    <DatasourceContainer>
+      <Control
+        {...datasourceControl}
+        name="datasource"
+        validationErrors={datasourceControl.validationErrors}
+        actions={actions}
+        formData={datasourceControl.mapStateToProps}
+      />
+      <div className="field-selections">
+        <input
+          type="text"
+          onChange={search}
+          className="form-control input-sm"
+          placeholder={t('Search Metrics & Columns')}
+        />
+        <Collapse
+          accordion
+          bordered={false}
+          defaultActiveKey={['column', 'metrics']}
+        >
+          <Collapse.Panel
+            header={<span className="header">{t('Columns')}</span>}
+            key="column"
+          >
+            <div className="field-length">
+              {`Showing ${columnSlice.length} of ${columns.length}`}
```

Comment:
I think you can use a t() string here too, there should be a way to add in template variables